text/0093-metric-instruments.md

```diff
@@ -0,0 +1,139 @@
+* Exported metric events include the instrument Descriptor, allowing exporters to interpret event values.
+
+In other words, these refinements serve to define the set of
+instruments.  Both users and exporters will deal in concrete kinds of
+instrument, these refinements are just for explaining their
+properties.
+
+OTEP 88 describes how we are meant to compute rate information from
+instruments having the Non-negative-rate refinement.  Temporal
+aggregation (over time) must be treated as a special case, compared
+with spatial aggregation (over labels).  The logic for computing rates
+depends on whether the Precomputed-sum refinement is present or not,
+which determines whether Delta or Cumulative values are being
+captured.
+
+OTEP 88 proposes that when adding new instruments, we specify
+instruments having a single purpose, each with a distinct set of
+refinements, and each with a carefully selected name for the
+properties of the instrument.
+
+OTEP 88 also also proposes to support language-specific specialization
+as well, to support built-in value types (e.g., timestamps).
+
+## Internal details
+
+The existing specification includes three instruments: Counter,
+Measure, and Observer.  In this proposal, the two foundational
+instruments, Measure and Observer (synchronous and asynchronous,
+respectively), become abstract names, in the sense that all
+synchronous instruments are instances of a Measure instrument and all
+asynchronous instruments are instances of an Observer instrument.  New
+instrument names are given to the unrestricted, unrefined versions of
+the foundational instruments.  The following table summarizes this
+proposal:
+
+| Existing name | Proposed name | Sync or Async | Refinements |
+| ------------- | ------------- | ------------- | ----------- |
+| Counter       | Counter       | Sync          | Sum-only, Non-negative, Non-negative-rate |
+| Measure       | Distribution  | Sync          | _None_  |
+|               | UpDownCounter | Sync          | Sum-only |
+|               | Timing        | Sync          | Non-negative, correct Duration units  |
+| Observer      | LastValueObserver | Async     | _None_ |
+|               | DeltaObserver | Async         | Sum-only, Non-negative, Non-negative-rate |
+|               | CumulativeObserver | Async    | Sum-only, Precomputed-sum, Non-negative-rate |
+
+### Synchronous instruments
+
+Two new synchronous instruments are introduced in this proposal, bringing the total to four.
+
+1. **Counter** remains unchanged.  It uses Sum aggregation by default.
+2. **Distribution** is an an unrefined Measure instrument.  Distribution accepts positive or negative values and uses MinMaxSumCount aggregation by default.
+3. **UpDownCounter** is a Sum-only instrument with no other refinements.  It supports capturing positive and negative changes to a sum (deltas).  UpDownCounter uses Sum aggregation by default.
```

Comment:
How come `UpDownCounter` cannot be treated as a refinement to a `Counter`?

Reply:
Counter has non-negative, which implies Rate support.  We think that's the proper default behavior, so `UpDownCounter` is actually less refined than `Counter`.

Reply:
`Non-negative-rate` is the refinement that defines rate support correct? Can there be a refinement like `up-down-rate` to `Counter` that would essentially be what `UpDownCounter` is capturing? Unless if I'm missing something.

Reply:
For a delta-oriented sum-only instrument, `non-negative` implies `non-negative rate`. I think of the `UpDownCounter` as lacking this `non-negative` refinement, thus lacking the `non-negative-rate` refinement, leaving it with only the `sum-only` refinement. 

Maybe we should be talking about a "property" of the instrument, which is not so much a refinement but a choice for sum-only instruments. Either they have the "delta" property or the "cumulative" property. `UpDownCounter` and `Counter` both have the "delta" property, as does `DeltaObserver`.  I think this "delta" property implies the `up-down-rate` you're asking about, but I don't see it as a refinement so much as a lack of the `non-negative` refinement.

(I'm not sure I'm helping...)

Reply:
I added a paragraph and a column to the table, hoping to explain how measurements are categorized as Delta, Cumulative, or Instantaneous based on which refinements are present. https://github.com/open-telemetry/oteps/pull/93/files#diff-998af0c6471c96e97ea74443a922ed92R59